pandas/core/internals.py

```diff
@@ -5440,6 +5449,14 @@ def is_uniform_join_units(join_units):
         len(join_units) > 1)
 
 
+def is_uniform_reindex(join_units):
+    return (
```

Comment:
Can you add a docstring to indicate what you mean with "uniform reindex"

I understand it as "this are uniform blocks that will keep being uniform after reindexing", correct? 
I am only wondering, does that not depend on the `fill_value` used? Or is this code path only taken in case of default filling (I don't think you can specify the filling with things like concat and merge, as opposed to reindex itself)

Reply:
> this are uniform blocks that will keep being uniform after reindexing

Exactly. That's the intent anyway...

It probably does depend on the `fill_value`. I need to better understand `get_empty_dtype_and_na`